Mathlib/Algebra/Order/GroupWithZero/WithZero.lean

```diff
@@ -78,3 +77,37 @@ instance {α : Type*} [Mul α] [Preorder α] [MulRightMono α] :
         dsimp only at h ⊢
         norm_cast at h ⊢
         exact mul_le_mul_right' h x
+
+section Units
+
+variable {α : Type*} [LinearOrderedCommGroupWithZero α]
+
+open WithZero
+
+/-- Given any linearly ordered commutative group with zero `α`, this is the order isomorphism
+between `WithZero αˣ` with `α`. -/
+@[simps!]
+def withZeroUnits_OrderIso : WithZero αˣ ≃o α where
+  __ := withZeroUnitsEquiv
+  map_rel_iff' {a b} := by
+    simp only [MulEquiv.toEquiv_eq_coe, EquivLike.coe_coe, withZeroUnitsEquiv_apply, recZeroCoe]
+    split
+    · exact ⟨fun _ ↦ WithZero.zero_le b, by simp⟩
+    · rcases b
+      · simpa using compareOfLessAndEq_eq_lt.mp rfl
+      rw [Units.val_le_val, ← WithZero.coe_le_coe]
+      rfl
+
+/-- Given any linearly ordered commutative group with zero `α`, this is the inclusion of
+`WithZero αˣ` into `α` as an ordered embedding. -/
+def withZeroUnits_OrderEmbedding : WithZero αˣ ↪o α := withZeroUnits_OrderIso.toOrderEmbedding
+
+@[simp]
+lemma withZeroUnits_OrderEmbedding_apply (x : WithZero αˣ) :
+    withZeroUnits_OrderEmbedding x = withZeroUnitsEquiv x := rfl
+
+lemma withZeroUnits_OrderEmbedding_mul (x y : WithZero αˣ) :
+    withZeroUnits_OrderEmbedding (x * y) = withZeroUnitsEquiv x * withZeroUnitsEquiv y := by
+  simp [map_mul]
+
+end Units
```

Comment:
I'm not sure these add anything that `withZeroUnits_OrderIso.toOrderEmbedding` doesn't already give you.